lib/ansible/plugins/action/win_reboot.py

```diff
@@ -155,11 +155,8 @@ def check_uptime():
                 display.vvv("attempting to get system uptime")
 
                 # override connection timeout from defaults to custom value
-                try:
-                    self._connection.set_options(direct={"connection_timeout": connect_timeout})
-                    self._connection._reset()
-                except AttributeError:
-                    display.warning("Connection plugin does not allow the connection timeout to be overridden")
+                self._connection.set_option("connection_timeout", connect_timeout)
+                self._connection._reset()
```

Comment:
This should probably be `reset` now (to work with non-winrm), and arguably both still need to be in an AttributeError catch block (for theoretical old plugins that don't extend ConnectionBase, though I don't know if we're bothering to support those. Regardless, `_reset` will fail on most plugins, and since `reset` is now on ConnectionBase...

Reply:
I was going to change it but reset is being used in SSH and _reset is specific in the task executor methods. I just wanted to fix this before making a change that could break something else